transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/acceptor/LimitsIT.java

```diff
@@ -0,0 +1,162 @@
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                            .acceptOption("ws.maximum.message.size", Integer.toString(DEFAULT_WEBSOCKET_MAXIMUM_MESSAGE_SIZE))
+                        .done()
+                    
+                        .service()
+                            .accept(URI.create("ws://localhost:8001/echoAuth"))
+                            .type("echo")
+                            .realmName("demo")
+                                .authorization()
+                                .requireRole("USER")
+                            .done()
+                        .done()
+                        .security()
+                            .realm()
+                                .name("demo")
+                                .description("Kaazing WebSocket Gateway Demo")
+                                .httpChallengeScheme("Basic")
+                                .sessionTimeout(String.valueOf(TEST_LIFETIME))
+                                .loginModule()
+                                    .type("class:org.kaazing.gateway.transport.wsn.auth.AsyncBasicLoginModule")
+                                    .success("requisite")
+                                    .option("roles", "USER")
+                                .done()
+                            .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    @Rule
+    public TestRule chain = createRuleChain(gateway, k3po);
+
+    @Test
+    @Specification({
+        "should.fail.binary.payload.length.126/handshake.request.and.frame"
+        })
+    public void shouldRefuseBinaryFrameWithPayloadLengthExceeding125() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "should.fail.binary.payload.length.131073/handshake.request.and.frame"
+        })
+    public void shouldRefuseBinaryFrameWithPayloadLengthExceeding128KiB() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "should.fail.text.payload.length.126/handshake.request.and.frame"
+        })
+    public void shouldRefuseTextFrameWithPayloadLengthExceeding125() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "should.fail.text.payload.length.131073/handshake.request.and.frame"
+        })
+    public void shouldRefuseTextFrameWithPayloadLengthExceeding128KiB() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "should.fail.max.lifetime.exceeded/handshake.request.and.frame"
+        })
+    public void shouldFailSendingMessageAfterSessionLifetimeIsUp() throws Exception {
+        k3po.start();
+        k3po.awaitBarrier("HANDSHAKE_COMPLETE");
+        Thread.sleep(3000);
```

Comment:
You should not need the sleep here since `k3po.awaitBarrier("CLOSE_RECEIVED")` will block until the barrier is notified by the request script, and that will only happen when the WebSocket session times out. You should add code to verify that the timing is about correct (stash System.currentTimeMillis instead of your Thread.sleep call and call it against after k3po.awaitBarrier("CLOSE_RECEIVED") and check the difference is at least say half of TEST_LIFETIME (converted to milliseconds). Don't make the test too straight because timing vary a lot between platforms.